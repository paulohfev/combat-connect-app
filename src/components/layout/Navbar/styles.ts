import { Theme } from '@mui/material'

const styles = {
  appBar: (theme: Theme) => ({
    height: theme.appBar.height,
  }),
  container: {
    alignItems: 'center',
    display: 'flex',
    height: '100%',
  },
  link: {
    alignItems: 'center',
    color: 'inherit',
    display: 'flex',
    textDecoration: 'none',
  },
  logoText: {
    marginLeft: '5px',
  },
}

export default styles
