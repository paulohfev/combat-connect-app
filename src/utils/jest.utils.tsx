import React from 'react'

import { ThemeOptions, ThemeProvider, createTheme } from '@mui/material'

import muiThemeConfig from '@/theme/mui-theme-config'
import { render } from '@testing-library/react'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(
    <ThemeProvider theme={createTheme(muiThemeConfig as ThemeOptions)}>{component}</ThemeProvider>
  )
}