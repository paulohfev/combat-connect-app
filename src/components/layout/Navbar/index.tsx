'use client'

import React from 'react'

import Link from 'next/link'

import { SportsMma } from '@mui/icons-material'
import { AppBar, Container, Typography } from '@mui/material'

import { APP_ROUTES } from '@/routes/app.routes'

import styles from './styles'

const Navbar: React.FC = () => {
  return (
    <AppBar position='fixed' sx={styles.appBar}>
      <Container sx={styles.container}>
        <Link href={APP_ROUTES.home} role='link' style={styles.link}>
          <SportsMma />
          <Typography variant='logo' sx={styles.logoText}>
            CombatConnect
          </Typography>
        </Link>
      </Container>
    </AppBar>
  )
}

export default Navbar
