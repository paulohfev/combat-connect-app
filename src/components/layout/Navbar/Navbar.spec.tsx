import React from 'react'

import { renderWithTheme } from '@/utils/jest.utils'
import { waitFor } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import Navbar from './index'

describe('Navbar', () => {
  it('should match the snapshot', () => {
    const navbar = renderWithTheme(<Navbar />)
    expect(navbar).toMatchSnapshot()
  })

  it('should redirect the user to the home page when the logo is clicked', () => {
    const navbar = renderWithTheme(<Navbar />)
    const logo = navbar.getByRole('link')

    act(() => {
      logo.click()
    })

    waitFor(() => {
      expect(window.location.pathname).toBe('/')
    })
  })
})
