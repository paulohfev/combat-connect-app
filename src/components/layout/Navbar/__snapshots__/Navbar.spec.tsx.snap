// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-17su868-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1nhcsch-MuiContainer-root"
        >
          <a
            href="/"
            role="link"
            style="align-items: center; display: flex; text-decoration: none;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SportsMmaIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 20c0 .55.45 1 1 1h8c.55 0 1-.45 1-1v-3H7v3zM18 7c-.55 0-1 .45-1 1V5c0-1.1-.9-2-2-2H7c-1.1 0-2 .9-2 2v5.8c0 .13.01.26.04.39l.8 4c.09.47.5.8.98.8h10.36c.45 0 .89-.36.98-.8l.8-4c.03-.13.04-.26.04-.39V8c0-.55-.45-1-1-1zm-3 3H7V7h8v3z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-logo css-6auo1l-MuiTypography-root"
            >
              CombatConnect
            </span>
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-17su868-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1nhcsch-MuiContainer-root"
      >
        <a
          href="/"
          role="link"
          style="align-items: center; display: flex; text-decoration: none;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SportsMmaIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 20c0 .55.45 1 1 1h8c.55 0 1-.45 1-1v-3H7v3zM18 7c-.55 0-1 .45-1 1V5c0-1.1-.9-2-2-2H7c-1.1 0-2 .9-2 2v5.8c0 .13.01.26.04.39l.8 4c.09.47.5.8.98.8h10.36c.45 0 .89-.36.98-.8l.8-4c.03-.13.04-.26.04-.39V8c0-.55-.45-1-1-1zm-3 3H7V7h8v3z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-logo css-6auo1l-MuiTypography-root"
          >
            CombatConnect
          </span>
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
